server/task/src/main/scala/org/apache/james/task/eventsourcing/TaskAggregate.scala

```diff
@@ -0,0 +1,71 @@
+/** **************************************************************
+  * Licensed to the Apache Software Foundation (ASF) under one   *
+  * or more contributor license agreements.  See the NOTICE file *
+  * distributed with this work for additional information        *
+  * regarding copyright ownership.  The ASF licenses this file   *
+  * to you under the Apache License, Version 2.0 (the            *
+  * "License"); you may not use this file except in compliance   *
+  * with the License.  You may obtain a copy of the License at   *
+  * *
+  * http://www.apache.org/licenses/LICENSE-2.0                 *
+  * *
+  * Unless required by applicable law or agreed to in writing,   *
+  * software distributed under the License is distributed on an  *
+  * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+  * KIND, either express or implied.  See the License for the    *
+  * specific language governing permissions and limitations      *
+  * under the License.                                           *
+  * ***************************************************************/
+package org.apache.james.task.eventsourcing
+
+import java.util
+
+import org.apache.james.eventsourcing.eventstore.History
+import org.apache.james.eventsourcing.{Event, EventId}
+import org.apache.james.task.Task
+import org.apache.james.task.Task.Result
+
+import scala.collection.JavaConverters._
+
+class TaskAggregate private(val aggregateId: TaskAggregateId, private val history: History) {
+
+  def create(task: Task): util.List[Event] = {
+    publishEvents(history,
+      Created(aggregateId, _, task))
+  }
+
+  private[eventsourcing] def start(): util.List[Event] = {
```

Comment:
does it make sense to accept this command (and emit an event) whatever is the aggregate state?

Reply:
I agree with you, we need a decision projection in our aggregate.